src/aggregation/agg_req_with_accessor.rs

```diff
@@ -121,29 +158,36 @@ pub(crate) fn get_aggs_with_accessor_and_validate(
 fn get_ff_reader_and_validate(
     reader: &SegmentReader,
     field_name: &str,
-) -> crate::Result<(DynamicFastFieldReader<u64>, Type)> {
+    multi: bool,
```

Comment:
I don't understand the multi argument here. 

What happens if multi == true and Cardinality is single valued?
Is this ok?

Can we comment on the function?
Shouldl the bool be of the type Cardinality instead? and call it cardinality too?

Reply:
I refactored the function and it accepts cardinality as parameter, I think it should be clear now

Reply:
Is this pushed?

Reply:
It should be now